### ISSUE TITLE
Feature request: Centering and numbering divs

### ISSUE THREAD

**Issue by ohepworthbell** [smashingboxes/OwlCarousel2#796]
It would be good if you could add a class such as 'before-1', 'before-2' and 'after-1', 'after-2', etc.  before and after the 'active' item in the carousel for additional styling, such that the carousel would appear as:

` <ul class="owl-carousel"> `
    ` <li class="item before-2">1</li> `
    ` <li class="item before-1">2</li> `
    ` <li class="item center active">3</li> `
    ` <li class="item after-1">4</li> `
    ` <li class="item after-2">5</li> `
    ` <li class="item after-3">6</li> `
    ` <li class="item after-4">7</li> `
` </ul> `

Then you could do things like having the opacity change the further a div is nested compared to the current (active) div. Doing it post-render as a plugin is a bit of a nightmare.

**Comment by shorstki** [comment]
I suppose you can use CSS n-th selector for this

**Comment by ohepworthbell** [comment]
Yes, but with cloned elements and responsive pages, this can be highly problematic. Also, nth child only really works for elements following after or as children of an anchor element (i.e. whilst you can have nth-child(n-2) it won't actually select a child element that precedes the first element because, well, it's the first element).

My feature request for the 'before-1' and 'before-2' class is mainly for preceding elements, than succeeding :)

Also, it overcomes problems with nth-selectors in IE8, etc. (yes, we should give up by now, but sometimes we don't have the luxury) and the various bugs in the 32-bit iOS devices and several other smartphone/tablets with nth-selectors.

**Comment by KennethAshley** [comment]
@webdesignhertfordshire This would be a great feature. We will add it to the feature list. You are more then welcome to submit a [pull request](https://github.com/smashingboxes/OwlCarousel2#pull-requests) for this feature.

**Comment by webdesignhertfordshire** [comment]
Hmm, I will have a look at the code and see if I can work out how to implement this. One thing I can envisage having trouble with is the 'items' option. Would you think it best that if 'items: 3' is enabled, for example, that the 'before-1' would effect one of the active items? And further from this, would you then have 3 different 'before-1' items, 3 different 'before-2' items, etc. ?

Or perhaps I am simply over-thinking this and it should literally be as simple as there is a centre, and then an 'item-distance' from the centre... I will look at this option for now, as it is certainly the most practical, and CSS can be used for styling the nth items anyway.

**Comment by KennethAshley** [comment]
@webdesignhertfordshire There will be some consideration that will have to go into the formatting of the 'before-' and 'after-'. If there are an even number of items shown, none are centered, and all are active. Odd numbers are more simple because you have a centered item for reference.
